### PR TITLE
enable defaults for custom attribute release using '' or 'default' key

### PR DESCRIPTION
**Pull request by leifj**
This also contains a small utility function that I will use to refactor some of my other pull requests later. There are tests covering this already so it should be safe as long as the tests work.

### CODE REVIEW

File: src/satosa/util.py [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 logger = logging.getLogger(__name__)
 
+def get_dict_defaults(d, *keys):
+    for key in keys:
+       d = d.get(key, d.get("", d.get("default", {})))
+    return d
```
Comment (johanlundberg) [inline src/satosa/util.py]: Am I understanding correctly that you want to accomplish this in other words?

def get_dict_defaults(d, *keys):
    for key in keys:
        if key in d:
            return d[key]
        elif '' in d or 'default' in d:
            return d.get('', d.get('default'))
    return dict()

Comment (leifj) [inline src/satosa/util.py]: No. I want to traverse the entire length of the "key chain" finding the most specific branch.

Comment (leifj) [inline src/satosa/util.py]: Basically this is the _config function I'm using in my two other PRs for attribute authz and attribute generation. Its a  pattern for config that I think @skoranda uses too in places - esp when you need to override behaviour specified by metadata on a per-Idp, per-sp+idp or globally

Comment (johanlundberg) [inline src/satosa/util.py]: Ok, I get it, you want to reuse d and the keys are in hierarchical order.

**Pull request merged**